### PR TITLE
Uninstall objects from charts in non-default namespaces

### PR DESCRIPTION
**Pull request by thedadams**
If the objects of a chart were installed in a non-default namespace
without the namespace being listed in the object's metadata, then the
uninstalled would seem successful but the helm is not able to find the
objects. The non-namespaced items would be removed, but any namespaced
items would be leftover.

It should be noted that the namespace provided allowed helm to find the
release secret, but would not namespace the call to delete the objects
in the release.

Issue:
https://github.com/rancher/rancher/issues/32179

### CODE REVIEW

File: pkg/catalogv2/system/system.go [hunk 1]
```diff
@@ -122,27 +122,26 @@ func (m *Manager) installCharts(charts map[desiredKey]map[string]interface{}) {
...
 
-	if len(releases) == 0 {
-		return nil
+	if wait {
+		if err = m.waitPodDone(op); err != nil {
+			return err
+		}
 	}
 
-	uninstall := action.NewUninstall(helmcfg)
```
Comment (StrongMonkey) [inline pkg/catalogv2/system/system.go:143]: Does it mean that when using this library to install, it doesn't clean up namespaced items? I am not sure if there is a difference between using library or using pod to uninstall.

Comment (thedadams) [reply pkg/catalogv2/system/system.go:143]: If the objects in the helm release had the namespace in their metadata, then it would clean them up.

However, if the objects did not have a namespace in their metadata and were installed in a non-default namespace, then objects were not getting cleaned up.

There is a way to use the library, but this seemed cleaner. If you think the library should continued to be used, then I can update this PR with that approach instead.

Comment (StrongMonkey) [reply pkg/catalogv2/system/system.go:143]: Ok fine by either way. Using library might be fast since it doesn't need to involve creating pods, but it is just one time uninstall so it shouldn't matter that much.

**Review by StrongMonkey — DISMISSED** [review]

**Review by nickgerace — DISMISSED** [review]

File: pkg/catalogv2/system/system.go (hunk 1 above)
Comment (StrongMonkey) [reply pkg/catalogv2/system/system.go:143]: @thedadams Actually I think we need to check if errors.isnotfound(). I am getting 
```
error syncing 'rancher-charts': handler bootstrap-charts: apps.catalog.cattle.io "rancher-operator" not found, requeuing
```
when trying to use your method. I happen to test something else with your PR.

Comment (thedadams) [reply pkg/catalogv2/system/system.go:143]: @StrongMonkey Added check for "not found" error.

**Review by StrongMonkey — APPROVED** [review]

File: pkg/catalogv2/system/system.go [hunk 2]
```diff
@@ -123,26 +123,19 @@ func (m *Manager) installCharts(charts map[desiredKey]map[string]interface{}) {
...
+		Timeout: &metav1.Duration{Duration: 5 * time.Minute},
+	})
 	if err != nil {
 		return err
 	}
 
-	if len(releases) == 0 {
-		return nil
+	op, err := m.operation.Uninstall(m.ctx, installUser, namespace, name, bytes.NewBuffer(uninstall))
+	if err != nil && !apierrors.IsNotFound(err) {
```
Comment (StrongMonkey) [inline pkg/catalogv2/system/system.go:134]: Actually this should be 
```
if err != nil {
  if apierrors.IsNotFound(err) {
    return nil
  }
  return err
}

Comment (thedadams) [reply pkg/catalogv2/system/system.go]: Done.

File: pkg/catalogv2/system/system.go [hunk 3]
```diff
@@ -123,26 +123,22 @@ func (m *Manager) installCharts(charts map[desiredKey]map[string]interface{}) {
 }
 
 func (m *Manager) Uninstall(namespace, name string) error {
-	helmcfg := &action.Configuration{}
```
Comment (cbron) [inline pkg/catalogv2/system/system.go:126]: Why remove all the list/check code ? The install method below still checks that inside `isPendingUninstall`, which I'm assuming is there so it doesn't error out. Must be some reason it existed.

**Review by cbron — APPROVED** [review]

File: pkg/catalogv2/system/system.go [hunk 4]
```diff
@@ -123,10 +123,19 @@ func (m *Manager) installCharts(charts map[desiredKey]map[string]interface{}) {
 }
 
 func (m *Manager) Uninstall(namespace, name string) error {
+	if ok, err := m.hasStatus(namespace, name, action.ListDeployed|action.ListFailed); err != nil {
```
Comment (rohitsakala) [inline pkg/catalogv2/system/system.go:126]: Why was this added ? this condition ?
